src/scalarstats.jl

```diff
@@ -260,37 +260,49 @@ variation(x) = ((m, s) = mean_and_std(x); s/m)
 realXcY(x::Real, y::Real) = x*y
 realXcY(x::Complex, y::Complex) = real(x)*real(y) + imag(x)*imag(y)
 
+
 """
-    sem(x)
+    sem(x[, weights::AbstractWeights]; [mean,])
+
+Return the standard error of the mean for a collection `x`. When using no weights, this is
+equal to the (sample) standard deviation divided by the sample size. If weights are used, 
+the variance of the sample mean is calculated as follows:
+
+* `AnalyticWeights`: Not implemented.
+* `FrequencyWeights`: ``\\frac{\\sum_{i=1}^n w_i (x_i - \\bar{x_i})^2}{(\\sum w_i) (\\sum w_i - 1)}``
+* `ProbabilityWeights`: ``\\frac{n}{n-1} \\frac{\\sum_{i=1}^n w_i^2 (x_i - \\bar{x_i})^2}{\\left( \\sum w_i \\right)^2}``
 
-Return the standard error of the mean of collection `x`,
-i.e. `sqrt(var(x, corrected=true) / length(x))`.
+The standard error is then the square root of the above quantities.
+
+# References
+
+Sribney, W. 1997. Probability weights, analytic weights, and summary statistics. Stata FAQ.
 """
-function sem(x)
-    y = iterate(x)
-    if y === nothing
-        T = eltype(x)
-        # Return the NaN of the type that we would get, had this collection
-        # contained any elements (this is consistent with std)
-        return oftype(sqrt((abs2(zero(T)) + abs2(zero(T)))/2), NaN)
+sem(x; mean=nothing) = sqrt(var(x; mean=mean, corrected=true) / length(x))
+
+function sem(x::AbstractArray, weights::UnitWeights; mean=nothing)
```

Comment:
No need to restrict this method to arrays since it uses the non-weights implementation:
```suggestion
function sem(x, weights::UnitWeights; mean=nothing)
```

Reply:
If we do this a test should be added, otherwise we might break it without noticing in the future.

Reply:
Sure :+1: